### ISSUE TITLE
Ability to expire Hydras when using Summoner

### ISSUE THREAD

**Issue by mlev** [robohydra/robohydra#13]
This is a feature request for the ability to expire or somehow clear out old Hydra instances created by a Summoner. I'm using robohydra to run as a mock api for some BDD tests and using a summoner enables us to run the tests in parallel and enable/disable the scenarios independently. However, once a test has run we will never reuse that Hydra.

At the moment it looks like the hydras are just stored in a plain object (e.g. https://github.com/robohydra/robohydra/blob/master/lib/robohydrasummoner.js#L20) so I'm guessing the memory footprint will gradually increase. We could replace this with something like https://github.com/isaacs/node-lru-cache and unused Hydras would drop off after a certain limit is exceeded.

**Comment by emanchado** [comment]
That's a good point. Something I don't like about the automatic clean-up of hydras is that they can contain state (dynamically created heads, active scenario, values of variables in certain plugins, etc) so removing a hydra and then using it again won't necessarily have the same result.

That's why I'd rather go for some explicit removal. I guess that, in your case, it would be enough to make an API call in the teardown method or similar.

I was actually considering making an API and maybe an admin UI to see the different active hydras, but I didn't know how much summoners would be used, and I thought it wasn't worth the effort.

What do you think about this approach, would that fit your usecase?

**Comment by mlev** [comment]
Yes - explicit removal would certainly meet our requirements. As you suggested we could delete it in the teardown for the tests. It would be nice to have it part of the REST api but a UI would be fine too (which we'd access with Selenium/Webdriver).

**Comment by emanchado** [comment]
You can have a look at the hydra-deletion branch and see if that seems enough:

https://github.com/robohydra/robohydra/tree/hydra-deletion

**Comment by mlev** [comment]
Looks pretty good to me. A couple of minor things I noticed were:
1) There is a button to delete the default Hydra but I'm guessing this doesn't really do anything. I thought it might delete and then recreate it but the Started time in the page footer doesn't change.
2) The response from the REST deletion is the the list of summoner hydras. It makes sense in the UI to return to this list but I wouldn't expect a restful deletion to return anything (apart from appropriate response code). https://github.com/robohydra/robohydra/commit/f2a7c8ac0417aedf4e93a49ff0dc604d28059d98#diff-845fcfc13ab990b982166f930797e544R515

Neither of these are big problems - in general looks great. Thanks

**Comment by emanchado** [comment]
Thanks for your feedback! About (1), it's actually a bug, the "Started" time is completely broken and I never noticed :-(

I've also changed the REST deletion response to use status code 204 No Content and an empty response body.

See the updated branch for the Started time fix.

**Comment by emanchado** [comment]
This is now in master BTW.

**Comment by mlev** [comment]
Brilliant - thanks.
